homeassistant/components/sensor/utility_meter.py

```diff
@@ -0,0 +1,264 @@
+        self._unit_of_measurement = None
+        self._period = meter_type
+        self._period_offset = meter_offset
+
+        if meter_type == HOURLY:
+            async_track_time_change(hass, self.async_reset_meter,
+                                    minute=meter_offset, second=0)
+        if meter_type == DAILY:
+            async_track_time_change(hass, self.async_reset_meter,
+                                    hour=meter_offset, minute=0, second=0)
+        elif meter_type in [WEEKLY, MONTHLY, YEARLY]:
+            async_track_time_change(hass, self._async_reset_meter,
+                                    hour=0, minute=0, second=0)
+
+    @callback
+    def async_reading(self, entity, old_state, new_state):
+        """Handle the sensor state changes."""
+        if old_state is None:
+            return
+
+        if self._unit_of_measurement is None:
+            self._unit_of_measurement = new_state.attributes.get(
+                ATTR_UNIT_OF_MEASUREMENT)
+
+        try:
+            diff = Decimal(new_state.state) - Decimal(old_state.state)
+
+            if diff < 0:
+                # Source sensor just rolled over for unknow reasons,
+                return
+            self._state += diff
+
+        except ValueError as err:
+            _LOGGER.warning("While processing state changes: %s", err)
+
+        self._hass.async_add_job(self.async_update_ha_state, True)
+
+    async def async_start_pause_meter(self):
+        """Start/Pause meter."""
+        if self._collecting is None:
+            # Start collecting
+            self._collecting = async_track_state_change(
+                self._hass, self._sensor_source_id, self.async_reading)
+        else:
+            # Pause collecting by cancel of async_track_state_change
+            self._collecting()
+            self._collecting = None
+
+        _LOGGER.debug("%s - %s - source <%s>", self._name,
+                      COLLECTING if self._collecting is not None
+                      else PAUSED, self._sensor_source_id)
+
+        self._hass.async_add_job(self.async_update_ha_state, True)
+
+    async def _async_reset_meter(self, event):
+        """Determine cycle - Helper function for larger then daily cycles."""
+        now = dt_util.now()
+        if self._period == WEEKLY and now.weekday() != self._period_offset:
+            return
+        if self._period == MONTHLY and\
+                now.day != (1 + self._period_offset):
+            return
+        if self._period == YEARLY and\
+                (now.month != (1 + self._period_offset) or now.day != 1):
+            return
+        await self.async_reset_meter(event)
+
+    async def async_reset_meter(self, event):
+        """Reset meter."""
+        _LOGGER.debug("Reset utility meter <%s>", self.entity_id)
+        self._last_reset = dt_util.now()
+        self._last_period = str(self._state)
+        self._state = 0
+        self._hass.async_add_job(self.async_update_ha_state, True)
```

Comment:
Why not `await self.async_update_ha_state(True)`?

Also, why `True`? This integration has no update function.

Reply:
Copy&Paste you know ;)